Config and quality of life

oheyjesse opened the pull request:
# Sssssssssssssss

This PR is for Quality Of Life fixes like better auto recompiling of JS files in dev, some basic linting setup, and a few additional little fixes.

# What do

- Add `watch` mode to webpack dev config. This watches for files, and recompiles when it sees a change. We still have to manually refresh the window - which is annoying. I was gonna go down this rabbithole but I didn't want to make too many big architectural changes at once

- Add ESLintrc - Config for ESLint - it's helpful for working in JS to have linting rules set up. I haven't gone through and formatted for this yet, though. I'd suggest we get to a point where our PR's are all in and merged up, then do a format based on the linter. 🤷‍♀ 

- Readme changes to reflect a few problems I ran into in Pythonland

Commit:
:wrench: Update webpack config to 'watch' files, remove 'main.js' (transpiled js file) from git

Commit:
:memo: Update README

Commit:
:wrench: Slight tweak to webpack watch (so it only watches in dev server mode)

Commit:
:wrench: Update requirements.txt with flake8 linter

Commit:
:lipstick: Add ESLint settings file for linty goodness

Commit:
:wrench: Webpack cache off in dev for CSS changes

mermop commented on .gitignore:
```diff
@@ -25,6 +25,7 @@ wheels/
 *.egg
 MANIFEST
 node_modules/
+frontend/static/frontend/main.js
```
There are a couple of things like this where they're committed both here and on #18.. can you tidy that up?

oheyjesse replied:
Yep, definitely. Looks like I didn't properly separate my branches. It was a little bit of "trying to keep the concerns separate, whilst finding a few changes as I went" with regards to this one. I'll go back over and try separate them out / put the right things in the right places.

mermop commented on webpack.config.js:
```diff
@@ -1,4 +1,8 @@
 module.exports = {
+  cache: false,
```
This is mostly duplicated in #18 but you've added different line 2s

mermop commented on README.md:
```diff
@@ -2,30 +2,49 @@
-To compile the JS, run `npm run dev`
+If you're actively developing frontend and would like the server to "watch" for changes, you can have webpack run a dev server, to watch for changes and recompile as you go!
+
+Frontend dev server: `npm run dev`
```
👍 Great documenting

mermop reviewed (approved):
Changes generally look fine but there's a bit of duplication with #18 - happy to merge but it might be a good idea to tidy those up!

Commit:
:art: Newline on gitignore

Commit:
:lipstick: Merge / Delete duplicate CSS files

oheyjesse commented:
Fixed this up - Made sure the infrastructure / webpack / linting / git / css changes are all in here. I removed the gitignore and webpack stuff from the layout PR - When I merge this, and rebase the updated master in the layout branch, those remaning "css file deleted" conflicts should well resolve fine.
